components/brave_wallet_ui/common/hooks/send.ts

```diff
@@ -194,12 +196,39 @@ export default function useSend () {
   const processSolanaAddress = React.useCallback((toAddressOrUrl: string) => {
     setToAddress(toAddressOrUrl)
 
-    if (toAddressOrUrl.toLowerCase() === selectedAccount?.address?.toLowerCase()) {
+    // Do nothing if value is an empty string
+    if (toAddressOrUrl === '') {
       setAddressWarning('')
+      setAddressError('')
+      return
+    }
+
+    // Check if value is the same as the sending address
+    if (toAddressOrUrl.toLowerCase() === selectedAccount?.address?.toLowerCase()) {
       setAddressError(getLocale('braveWalletSameAddressError'))
       return
     }
 
+    // Check if value is a Tokens Contract Address
+    if (fullTokenList.some(token => token.contractAddress.toLowerCase() === toAddressOrUrl.toLowerCase())) {
+      setAddressError(getLocale('braveWalletContractAddressError'))
+      return
+    }
+
+    // Check if value is a Base58 Encoded Solana Pubkey
+    isBase58EncodedSolanaPubkey(toAddressOrUrl).then((value: IsBase58EncodedSolanaPubkeyReturnInfo) => {
+      const { result } = value
+
+      if (result) {
+        setAddressError('')
+        return
+      }
+
+      if (!result) {
+        setAddressError(getLocale('braveWalletNotValidSolAddress'))
+      }
+    }).catch(e => console.log(e))
+
     setAddressWarning('')
```

Comment:
these probably should go in a `.finally` of `isBase58EncodedSolanaPubkey` to prevent race conditions